subprojects/launcher/src/main/java/org/gradle/launcher/daemon/configuration/ForegroundDaemonConfiguration.java

```diff
@@ -23,7 +23,6 @@
 public class ForegroundDaemonConfiguration extends DefaultDaemonServerConfiguration {
     public ForegroundDaemonConfiguration(String daemonUid, File daemonBaseDir, int idleTimeoutMs, int periodicCheckIntervalMs) {
         // Foreground daemon cannot be 'told' what's his startup options as the client sits in the same process so we will infer the jvm opts from the inputArguments()
-        // Simplification, we will make the foreground daemon interested only in managed jvm args
```

Comment:
This was not a great choice, since it meant that foreground daemons were pretty much never reused.